### ISSUE TITLE
Features needed in CMSSW_10_1_X for 90 m beta* Totem run

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#23275]
I collect here the list of the needed backports to CMSSW_10_1_X from the master release to support the version to be used for Totem run with beta* = 90 m planned for the second part of June.

I indicate the master version first, and then the backport to 10_1_X (where already existing)

For HLT:

Add layer counting capability to PixelActivityFilter
- [ ]   #22951 (merged) -> #22952 

For geometry and local reconstruction

Totem Timing detectors: updated mapping for dedicated run
- [ ] #23198 (merged) -> ?

CTPPS Strip detectors: updated mapping for dedicated run
- [ ] #23119 (merged) -> ?

CTPPS: TotemTimingDigi, totemTimingRawToDigi and mapping for the new Timing detector in Vertical RPs
- [ ] #22796 (merged) -> 

Totem Timing RecHit Producer
- [ ] #23101 (merged) ->

For tracking tuning:

Tracking hgh beta star 2018
- [ ] #23238 (under RECO review)

For DQM:

Totem timing DQM
- [ ] #22915 (merged) ->

CTPPS: DQM update
- [ ] #23025 (merged) -> ?

**Comment by fabiocos** [comment]
@nminafra @slava77 @perrotta FYI

**Comment by fabozzi** [comment]
@slava77 @perrotta 
We had actually some questions for you in the thread: "Re: preparation of TOTEM run: data relval"
i.e. which era and customise to be used in the cmsDriver of data relval.

I see in #23238 that era Run2_2018_highBetaStar has been defined.
Not sure about customise (RecoTLR.customisePostEra_Run2_2018 is OK?).

Another question was for Alca team @arunhep @lpernie about the GT to use from autoCond. 

After these ingredients are confirmed, then the data relval can be setup, since we have already received info about run/lumisec to use:
Run 314890 (LS= 500-926) -- for central tracking tests
Run 314276 (any LS range) -- for RomanPot system (new timing detectors
are included, central tracker NOT IN RUN)
dataset = ZeroBias

**Comment by slava77** [comment]
On 5/27/18 9:02 AM, fabozzi wrote:
> @slava77 <https://github.com/slava77> @perrotta
> <https://github.com/perrotta>
> We had actually some questions for you in the thread: "Re: preparation
> of TOTEM run: data relval"
> i.e. which era and customise to be used in the cmsDriver of data relval.
> 
> I see in #23238 <https://github.com/cms-sw/cmssw/pull/23238> that era
> Run2_2018_highBetaStar has been defined.
> Not sure about customise (RecoTLR.customisePostEra_Run2_2018 is OK?).

please use Run2_2018_highBetaStar
I suggest to use the same customise for 2018

> 
> Another question was for Alca team @arunhep <https://github.com/arunhep>
> @lpernie <https://github.com/lpernie> about the GT to use from autoCond.
> 
> After these ingredients are confirmed, then the data relval can be
> setup, since we have already received info about run/lumisec to use:
> Run 314890 (LS= 500-926) -- for central tracking tests
> Run 314276 (any LS range) -- for RomanPot system (new timing detectors
> are included, central tracker NOT IN RUN)
> dataset = ZeroBias
>

**Comment by slava77** [comment]
@fabiocos 

a backport of #23349 will be needed for 10_1_X as well.
This will come after a backport of #23238 is in 10_1_X.

**Comment by forthommel** [comment]
@fabiocos 
A backport of #22796, #23101 and #23198 has been opened as #23352 
We will follow up with another backport for #22915.

**Comment by fabiocos** [comment]
@forthommel do I understand correctly that #23352 includes also #23119?

**Comment by perrotta** [comment]
It doesn't seem so, but it probably should. @forthommel?

Fabio Cossutti <notifications@github.com> ha scritto:

**Comment by forthommel** [comment]
@fabiocos, @perrotta : thanks a lot for checking! Indeed, #23119 is not included in these two PRs (tackles the strips detectors, not the TOTEM vertical timing detectors).
I am now launching an independent PR for the backport of this latter.

**Comment by forthommel** [comment]
#23356 was just launched to backport #23119 onto `CMSSW_10_1_X`.

**Comment by arunhep** [comment]
@fabozzi 

> Another question was for Alca team @arunhep @lpernie about the GT to use from autoCond.
> After these ingredients are confirmed, then the data relval can be setup, since we have already received info about run/lumisec to use:
> Run 314890 (LS= 500-926) -- for central tracking tests
> Run 314276 (any LS range) -- for RomanPot system (new timing detectors
> are included, central tracker NOT IN RUN)
> dataset = ZeroBias

how about using prompt conditions?
in that case you can use this key : https://github.com/cms-sw/cmssw/blob/master/Configuration/AlCa/python/autoCond.py#L33

**Comment by fabozzi** [comment]
I have created PR #23388 to deploy the 2 data workflows.
At the moment the workflow on run 314276 uses MinimumBias since no ZeroBias PD is available for that run. I am waiting confirmation from Totem experts on which dataset should be actually used.

**Comment by arunhep** [comment]
may be totem experts can clarify one more thing. do they need this updated geometry : https://hypernews.cern.ch/HyperNews/CMS/get/calibrations/3667.html in GT for this relval?

**Comment by forthommel** [comment]
@arunhep Yes, this geometry tag is required to pick up the timing detectors part.

**Comment by nminafra** [comment]
@fabozzi Any of the ZeroBiasn with n = 1 to 10 is fine.
Also MinimumBias should be fine, but we usually use ZeroBias1.
Thanks

**Comment by fabiocos** [comment]
CMSSW_10_1_7 has been uploaded and announced, this closes the list of requests received for the preparation of the Totem run
